### DISCUSSION TITLE
BranchPythonOperator Task id 'select_task' has already been added to the DAG

### DISCUSSION THREAD

**Discussion by FyZzyss** [apache/airflow#18057] [General] [answered]
I want to add optional tasks using BranchPythonOperator after the loop. I get an error `airflow.exceptions.DuplicateTaskIdFound: Task id 'select_task' has already been added to the DAG
`
Code:
```
def select_next_task():
    # if release_date and activate_date:
    if True:
        return 'add_new_coupons'
    return 'replace_temp_tables'
__________________________________________________
@dag(default_args=default_args, schedule_interval=None, start_date=days_ago(2), tags=['dags'])
    for number in range(total_task_chunks):
        get_coupons = get_coupons_info(referrals[str(number)], dates['release_date'], dates['activate_date'])
        select_task = BranchPythonOperator(
            task_id='select_task',
            python_callable=select_next_task,
        )
        get_coupons.set_downstream(select_task)
```

How to fix this?

**Comment by josh-fell** [comment, marked as answer]
Are you trying to create multiple `BranchPythonOperator` tasks or have the different "get_coupons_info" tasks have a downstream dependency of the same `BranchPythonOperator` task?

If you are trying to create multiple "select_task" tasks, you just need to make sure the `task_id` value is unique for the DAG.

If you want all of the "get_coupon_info" tasks (assuming this is a task-decorated function using the TaskFlow API) to have a downstream dependency on "select_task", then bring the "select_task" instantiation outside of the loop.  Something like this:
```python
    select_task = BranchPythonOperator(
        task_id='select_task',
        python_callable=select_next_task,
    )
    
    for number in range(total_task_chunks):
        get_coupons = get_coupons_info(referrals[str(number)], dates['release_date'], dates['activate_date'])
        get_coupons.set_downstream(select_task)
```
This way you are not trying to create several instances of the same "select_task" task.

**Reply by FyZzyss** [reply]
Oh, my bad. Thank you. I wanted downstream dependency on "select_task"
